connection-service/new-admin-vm/Install-ConnectionServer.ps1

```diff
@@ -917,44 +937,21 @@ isMultiFactorAuthenticate=$isMfa
-
-                        # Authenticate with SSL - trusting all certificates
-                        $sslstream = new-object System.Net.Security.SslStream -ArgumentList $tcpclient.GetStream(),$false,{$true}
-
-                        $sslstream.AuthenticateAsClient($hostname)
-                        $cert =  [System.Security.Cryptography.X509Certificates.X509Certificate2]($sslstream.remotecertificate)
-                        Write-Host "Found Certificate for $hostname, looking for Issuer Certificate"
-
-                        $chain = New-Object -TypeName System.Security.Cryptography.X509Certificates.X509Chain
-                        # Build the certificate chain from the file certificate
-                        $chain.Build($cert)                        
-                        $listOfCertificates = ($chain.ChainElements | ForEach-Object {$_.Certificate})
-                        
-                        #last one is the root in chain
-                        $caCert=$listOfCertificates[-1]
-                        $content = @(
-                            '-----BEGIN CERTIFICATE-----'
-                            [System.Convert]::ToBase64String($caCert.RawData, 'InsertLineBreaks')
-                            '-----END CERTIFICATE-----'
-                        )
-                       
-                        $content | Out-File -FilePath "$env:systemdrive\$issuerCertFileName" -Encoding ascii
-                        $foundCert=$true
+                        Start-Process C:\OpenSSL-Win64\bin\openssl.exe -ArgumentList "s_client -showcerts -connect ${hostname}:${port}" -RedirectStandardOutput "$env:systemdrive\certInfo.txt"
+                        # Wait a bit to ensure that the previous command completes
+                        Sleep 10
+                        $openSSLOutput = Get-Content "$env:systemdrive\certInfo.txt" -Raw
+                        $certMatches = $openSSLOutput | Select-String '(?smi)(-----BEGIN CERTIFICATE-----[^-]+-----END CERTIFICATE-----)' -AllMatches | % {$_.Matches}
+                        if ($certMatches.Count) {
+                            #last one is the root in chain
+                            $certMatches[-1].Value | Out-File -FilePath "$env:systemdrive\$issuerCertFileName" -Encoding ascii
+                            $foundCert=$true
+                        }
                     } catch {
-                        Write-Host "Failed to retrieve issuer certificate from $hostname`:$port because $_"
+                        Write-Host "Failed to retrieve issuer certificate from ${hostname}:${port} because $_"
                     } finally {
-                        #cleanup
-                        if ($sslStream) {
-                            $sslstream.close()  | Out-Null
-                        }
-                        if ($tcpclient) {
-                            $tcpclient.close()  | Out-Null
-                        }
+                        Taskkill /F /IM "openssl.exe"
```

Comment:
I guess it's usually already dead by now...?

Reply:
I couldn't figure out how to auto-quit :(, shell stays open until we kill it

Reply:
Just make it 30 or something...